src/analysis/special_functions/trigonometric.lean

```diff
@@ -1062,89 +1063,113 @@ by simp [two_mul, sin_add]
 @[simp] lemma cos_two_pi : cos (2 * π) = 1 :=
 by simp [two_mul, cos_add]
 
-lemma sin_nat_mul_pi (n : ℕ) : sin (n * π) = 0 :=
-by induction n; simp [add_mul, sin_add, *]
+lemma sin_antiperiodic : function.antiperiodic sin π :=
+by simp [sin_add]
 
-lemma sin_int_mul_pi (n : ℤ) : sin (n * π) = 0 :=
-by cases n; simp [add_mul, sin_add, *, sin_nat_mul_pi]
+lemma sin_periodic : function.periodic sin (2 * π) :=
+sin_antiperiodic.periodic
 
-lemma cos_nat_mul_two_pi (n : ℕ) : cos (n * (2 * π)) = 1 :=
-by induction n; simp [*, mul_add, cos_add, add_mul, cos_two_pi, sin_two_pi]
+lemma sin_add_pi (x : ℝ) : sin (x + π) = -sin x :=
+sin_antiperiodic x
 
-lemma cos_int_mul_two_pi (n : ℤ) : cos (n * (2 * π)) = 1 :=
-by cases n; simp only [cos_nat_mul_two_pi, int.of_nat_eq_coe, int.neg_succ_of_nat_coe,
-                      int.cast_coe_nat, int.cast_neg, ← neg_mul_eq_neg_mul, cos_neg]
+lemma sin_add_two_pi (x : ℝ) : sin (x + 2 * π) = sin x :=
+sin_periodic x
 
-lemma sin_add_pi (x : ℝ) : sin (x + π) = -sin x :=
-by simp [sin_add]
+lemma sin_sub_pi (x : ℝ) : sin (x - π) = -sin x :=
+sin_antiperiodic.sub_eq x
 
-lemma sin_add_int_mul_two_pi (x : ℝ) (n : ℤ) : sin (x + n * (2 * π)) = sin x :=
-begin
-  rw [sin_add, cos_int_mul_two_pi, ← mul_assoc],
-  rw_mod_cast sin_int_mul_pi (n*2),
-  simp,
-end
+lemma sin_sub_two_pi (x : ℝ) : sin (x - 2 * π) = sin x :=
+sin_periodic.sub_eq x
 
-lemma sin_sub_int_mul_two_pi (x : ℝ) (n : ℤ) : sin (x - n * (2 * π)) = sin x :=
-by simpa using sin_add_int_mul_two_pi x (-n)
+lemma sin_pi_sub (x : ℝ) : sin (π - x) = sin x :=
+by simpa only [sub_eq_neg_add, sin_neg, neg_neg] using sin_antiperiodic (-x)
```

Comment:
This looks like it could be a general lemma about `antiperiodic f n` and `f (n - x) = f x`